hpx/runtime/parcelset/parcel.hpp

```diff
@@ -149,182 +104,109 @@ namespace hpx { namespace parcelset
             data()
               : start_time_(0),
                 creation_time_(util::high_resolution_timer::now()),
-                has_source_id_(false)
+                source_id_(naming::invalid_gid),
+                dest_(naming::invalid_gid)
+            {}
+
+            data(naming::gid_type&& dest, naming::address&& addr)
+              : start_time_(0),
+                creation_time_(util::high_resolution_timer::now()),
+                source_id_(naming::invalid_gid),
+                dest_(std::move(dest)),
+                addr_(std::move(addr))
             {}
 
             data(data && rhs)
-              : parcel_id_(rhs.parcel_id_),
-                start_time_(rhs.start_time_),
+              : start_time_(rhs.start_time_),
                 creation_time_(rhs.creation_time_),
-                has_source_id_(rhs.has_source_id_)
+                source_id_(std::move(rhs.source_id_)),
+                dest_(std::move(rhs.dest_)),
+                addr_(std::move(rhs.addr_))
             {
-                rhs.parcel_id_ = naming::invalid_gid;
                 rhs.start_time_ = 0;
                 rhs.creation_time_ = 0;
-                rhs.has_source_id_ = false;
+                rhs.creation_time_ = 0;
+                rhs.source_id_ = naming::invalid_gid;
+                rhs.dest_ = naming::invalid_gid;
+                rhs.addr_ = naming::address();
             }
 
             data& operator=(data && rhs)
             {
-                parcel_id_ = rhs.parcel_id_;
                 start_time_ = rhs.start_time_;
                 creation_time_ = rhs.creation_time_;
-                has_source_id_ = rhs.has_source_id_;
+                source_id_ = std::move(rhs.source_id_);
+                dest_ = std::move(rhs.dest_);
+                addr_ = std::move(rhs.addr_);
 
-                rhs.parcel_id_ = naming::invalid_gid;
                 rhs.start_time_ = 0;
                 rhs.creation_time_ = 0;
-                rhs.has_source_id_ = false;
+                rhs.source_id_ = naming::invalid_gid;
+                rhs.dest_ = naming::invalid_gid;
+                rhs.addr_ = naming::address();
                 return *this;
             }
 
             template <typename Archive>
             void serialize(Archive &ar, unsigned)
             {
-                ar & parcel_id_;
                 ar & start_time_;
                 ar & creation_time_;
-                ar & has_source_id_;
+                ar & source_id_;
+                ar & dest_;
+                ar & addr_;
             }
 
-            naming::gid_type parcel_id_;
```

Comment:
Am 03.09.2016 10:34 nachm. schrieb "Hartmut Kaiser" <
notifications@github.com>:
>
> In hpx/runtime/parcelset/parcel.hpp:
>
> > } > > - naming::gid_type parcel_id_;
>
> It is necessary for trace tools which track things across the network.
Please re-add this. The cost should be neglectable.

It's a 128 bit number that needs to be transferred in addition. Significant
for small parcels.

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.